src/openapi-generator.ts

```diff
@@ -333,7 +333,7 @@ export class OpenAPIGenerator {
     const required =
       !this.isOptionalSchema(zodSchema) && !zodSchema.isNullable();
 
-    const schema = this.generateSimpleSchema(zodSchema);
+    const schema = this.generateSchemaDefinition(zodSchema);
```

Comment:
I don't think I can agree with this one. Right now this change would enforce a schema being defined into `schemaRefs` and would then be saved in `paramRefs` (in `generateParameterDefinition`). That was the overall idea behind having a `generateSimpleSchema` that does not bother "saving the result ref".

Reply:
Oh I see I see, let me rethink this :)